From d99f791071ea5a2a1acbd270f5ef9505dd552c05 Mon Sep 17 00:00:00 2001
From: Rafael Palomar <rafael.palomar@ous-research.no>
Date: Fri, 20 Feb 2026 21:47:58 +0100
Subject: [PATCH] ENH: Install CMake template files (.h.in, .cxx.in) alongside
 cmake modules

vtkSlicerObjectFactory.h.in and vtkSlicerObjectFactory.cxx.in are
required by SlicerConfigureDisplayableManagerObjectFactory.cmake when
building standalone loadable modules (e.g. Colors) that include a
displayable manager component.  The previous glob only captured *.cmake
files so these templates were missing from the install tree.
---
 CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 820cfc36da..99e6a20515 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1414,7 +1414,10 @@ endif()
 #-----------------------------------------------------------------------------
 if(NOT Slicer_INSTALL_NO_DEVELOPMENT)
 
-  file(GLOB cmake_files "${CMAKE_CURRENT_SOURCE_DIR}/CMake/*.cmake")
+  file(GLOB cmake_files
+    "${CMAKE_CURRENT_SOURCE_DIR}/CMake/*.cmake"
+    "${CMAKE_CURRENT_SOURCE_DIR}/CMake/*.h.in"
+    "${CMAKE_CURRENT_SOURCE_DIR}/CMake/*.cxx.in")
   install(
     FILES ${cmake_files}
     DESTINATION ${Slicer_INSTALL_LIB_DIR}/CMake COMPONENT Development
-- 
2.54.0

